packages/graphql/src/schema/make-augmented-schema.ts

```diff
@@ -966,6 +974,8 @@ function makeAugmentedSchema(
                     name: `${connectionField.typeMeta.name}Options`,
                     fields: {
                         sort: connectionSort.NonNull.List,
+                        first: "Int",
+                        after: "String",
```

Comment:
Good question. So I think the arguments need to be moved up to be direct arguments of the field. There is nothing in the specification, but if you check out the relay-compiler (used with the Relay client), it'll [throw an error](https://github.com/facebook/relay/blob/main/packages/relay-compiler/transforms/ConnectionTransform.js#L698) if it can't find arguments directly on the field.

I also could not find any examples where the arguments were nested. A person once raised the [issue](https://github.com/facebook/relay/issues/3285) but there was no comment or resolution from FB and the person closed it without any advice.

I'll move them out of the options object on next go. Any thoughts if I should hoist the sort argument as well so everything is directly on the field?

Reply:
Sure, so definitely hoist up `first` and `after`.

`sort` is tricky! If we hoist it up, it becomes inconsistent with what is on a "normal" relationship field, but you could argue that it's already inconsistent what with `first` and `after` at the root. But then if we leave it, it's sat there in `options` all by itself which doesn't make an awful lot of sense!

My gut says also hoist it up and bin off the `ConnectionOptions` types, as `where` and `sort` go quite well together. However, would like to get some thoughts off @danstarns and/or @oskarhane before we make a decision here.

Reply:
Had a chat with @danstarns, I think go ahead and also hoist up `sort` as well, as we're already inconsistent with where the pagination arguments sit. Also means we can get rid of the `ConnectionOptions` type which is nice. 

You're also going to want to merge from the upstream 2.0.0 branch as we've merged a couple of branches in and it looks like the tests are a bit broken again.